job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PullRequestPublisherContext.groovy

```diff
@@ -23,4 +25,12 @@ class PullRequestPublisherContext implements Context {
     void disallowOwnCode(boolean disallowOwnCode = true) {
         this.disallowOwnCode = disallowOwnCode
     }
+
+    void failOnNonMerge(boolean failOnNonMerge = true) {
```

Comment:
Add a `@RequiresPlugin` annotation with a minimum version. Either use the latest version of the ghprb plugin or figure out the version in which these settings have been added. Also add a `@since` annotation with a version set to the next release.